### PR TITLE
Initial commit of iNav protocol

### PR DESCRIPTION
**Pull request by martinbudden**
Added new iNav protocol, as to be included in https://github.com/iNavFlight/inav release 1.2

### CODE REVIEW

File: src/protocol/inav_nrf24l01.c [hunk 1]
```diff
@@ -0,0 +1,814 @@
...
+        // them by their numbers
+        // It's all magic, eavesdropped from real transfer and not even from the
+        // data sheet - it has slightly different values
+        NRF24L01_WriteRegisterMulti(0x00, (u8 *) "\x40\x4B\x01\xE2", 4);
+        NRF24L01_WriteRegisterMulti(0x01, (u8 *) "\xC0\x4B\x00\x00", 4);
+        NRF24L01_WriteRegisterMulti(0x02, (u8 *) "\xD0\xFC\x8C\x02", 4);
+        NRF24L01_WriteRegisterMulti(0x03, (u8 *) "\x99\x00\x39\x21", 4);
+        NRF24L01_WriteRegisterMulti(0x04, (u8 *) "\xF9\x96\x82\x1B", 4);
+        NRF24L01_WriteRegisterMulti(0x05, (u8 *) "\x24\x06\x7F\xA6", 4);
+        NRF24L01_WriteRegisterMulti(0x06, (u8 *) "\x00\x00\x00\x00", 4);
```
Comment (victzh) [inline src/protocol/inav_nrf24l01.c]: Please remove initialization of Beken registers 0x06 -0x0b with zeros. It's an artifact of decoding of sloppy initialization code from some early TX.

Comment (martinbudden) [inline src/protocol/inav_nrf24l01.c]: OK, I'll remove.

I don't really understand the point of this Beken initialisation - do people mod their Devo transmitters by adding in Beken chips?

Comment (goebish) [inline src/protocol/inav_nrf24l01.c]: hexfet does ;)

**Comment by martinbudden** [comment]
@goebish , I'm fairly happy with this protocol now. What needs to happen to get it into the deviationtx nightlies?

**Comment by goebish** [comment]
It needs to be approved by a couple of people.
No objection on my side as this won't break anything anyway.

**Comment by martinbudden** [comment]
Great, thanks for merging this in.

**Pull request merged**